### PR TITLE
SITE-23: Fixing form error not showing and making Shopping Cart Widget more customizable

### PR DESCRIPTION
**Pull request by DemeSzabolcs**
SITE-23

### CODE REVIEW

**Review by wAsnk — APPROVED** [review]

**Review by wAsnk — CHANGES_REQUESTED** [review]

**Review by wAsnk — APPROVED** [review]

**Comment by wAsnk** [comment]
@DAud-IcI Could you review and merge this?

**Comment by DAud-IcI** [comment]
Ok, I will review it soon.

File: src/Modules/OrchardCore.Commerce/Assets/Scripts/stripe-card-form.js [hunk 1]
```diff
@@ -85,13 +85,13 @@ window.stripeCardForm = function stripeCardForm(stripe, antiForgeryToken, urlPre
...
 
-        document.getElementById('StripePaymentPart_PaymentIntentId_Text').value = result.paymentIntent.id;
-
         // Show error in payment form.
         if (error) {
             displayError(error);
         }
         else {
+            document.getElementById('StripePaymentPart_PaymentIntentId_Text').value = result.paymentIntent.id;
+
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Assets/Scripts/stripe-card-form.js:94]: Would be better to simply `return` in the `if (error)`. Same below.

File: src/Modules/OrchardCore.Commerce/Views/ShoppingCartWidgetPart.cshtml [hunk 2]
```diff
@@ -1,9 +1,5 @@
 @model ShoppingCartWidgetPartViewModel
 
-<style asp-src="~/OrchardCore.Commerce/css/shopping-cart-widget.min.css"></style>
+@Model.ItemCount
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Views/ShoppingCartWidgetPart.cshtml:3]: The purpose of this is not really clear to me. I can't imagine a scenario where I'd want to customize just the text inside the circle. I could see wanting to customize the circle itself (e.g. adding a dropdown or hiding it). So I think the `<div class="shopping-cart-item-count text-white bg-danger text-center">` should be moved into this shape.

File: src/Modules/OrchardCore.Commerce/Views/Widget__ShoppingCartWidget.cshtml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<style asp-src="~/OrchardCore.Commerce/css/shopping-cart-widget.min.css"></style>
+
+<div class="shopping-cart-widget">
+    <a asp-action="Index" asp-controller="ShoppingCart" asp-area="OrchardCore.Commerce" class="shopping-cart-link nav-link d-flex">
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Views/Widget__ShoppingCartWidget.cshtml:4]: I think you don't need `asp-area` if the link points to the same module.

**Review by DAud-IcI — CHANGES_REQUESTED** [review]

**Pull request merged**